### ISSUE TITLE
PEP-8 compliancy: Indentation

### ISSUE THREAD

**Issue by ngiese** [SoFiA-Admin/SoFiA#147]
The indentation in our python code has been bugging be for years. In all other code, that I write apart from SoFiA, I use 4 spaces for indentation and every time I want to make changes in SoFiA, I have to change the settings of my text editor.
Since the style guide for python recommends the use of 4 spaces per indentation level (https://www.python.org/dev/peps/pep-0008/#indentation), I suggest that we start replacing our tabs to conform with this. I know, that we have had this conversation a few years back and the suggestion was to adjust our editors to replace tabs with spaces, but that still does not make our code PEP-8 compliant.

**Comment by SoFiA-Admin** [comment]
I would argue against this change for a number of reasons:

* Firstly, the argument about text editor settings is not really useful. I always use 1 tab per indentation level and likewise would have to change my text editor settings if we decided to switch to spaces instead.
* 4 spaces will occupy four times as much memory as 1 tab, so we would essentially inflate the file size of the SoFiA source code by about 10-20% without gaining anything.
* There is global consensus on using 1 tab per indentation level, while there is no consensus on how many spaces to use (PEP-8 recommends 4). Before we switched to tabs, I had to clean up indentation in SoFiA’s code in quite a few cases because someone had chosen a different number of spaces in individual code blocks. I don’t really want to get back to that situation, as it means extra work.
* By using tabs, we would be much less likely to get it wrong by accident, e.g. by unintentionally using 5 instead of 4 spaces (this actually occurred in the SoFiA code in the past), as any error in the number of tabs would be immediately obvious.
* Lastly, using tabs solely for indentation and spaces solely for separation is the most semantically meaningful choice in my opinion, as the tabulator key on typewriters was originally invented to move the carriage to the next column.

In summary, the PEP-8 recommendation for indentation is an unfortunate choice in my opinion and I would advise against complying with this specific part of PEP-8. In fact, I sincerely hope that all major software developers are using tabs in their source code, as otherwise they would potentially be shipping tens of MB of empty space to their users that would simply be wasting precious disk space.

**Comment by SoFiA-Admin** [comment]
As discussed at our recent meetings, the new [SoFiA coding style guide](https://github.com/SoFiA-Admin/SoFiA/wiki/Coding-Style-Guide) is now available on the wiki, and this issue can therefore be closed.
